Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -2143,33 +2143,38 @@ def fetch_emails_as_incidents(client: EWSClient, last_run):
     :return:
     """
     last_run = get_last_run(client, last_run)
-
+    excluded_ids = set(last_run.get(LAST_RUN_IDS))
     try:
         last_emails = fetch_last_emails(
             client,
             client.folder_name,
             last_run.get(LAST_RUN_TIME),
-            last_run.get(LAST_RUN_IDS),
+            excluded_ids,
         )
 
-        ids = deque(
-            last_run.get(LAST_RUN_IDS, []), maxlen=client.last_run_ids_queue_size
-        )
         incidents = []
         incident: Dict[str, str] = {}
+        demisto.debug(f'{APP_NAME} - Started fetch with {len(last_emails)}')
+        current_fetch_ids = set()
         for item in last_emails:
             if item.message_id:
-                ids.append(item.message_id)
+                current_fetch_ids.add(item.message_id)
                 incident = parse_incident_from_item(item)
                 incidents.append(incident)
 
                 if len(incidents) >= client.max_fetch:
                     break
 
+        demisto.debug(f'{APP_NAME} - ending fetch - got {len(incidents)} incidents.')
         last_run_time = incident.get("occurred", last_run.get(LAST_RUN_TIME))
         if isinstance(last_run_time, EWSDateTime):
             last_run_time = last_run_time.ewsformat()
 
+        if last_run_time > LAST_RUN_TIME:
+            ids = current_fetch_ids
+        else:
+            ids = current_fetch_ids | excluded_ids
```

Comment:
add a comment that explains the logic here.It is not clear to me what this code does and why